### PR TITLE
v1.1: Long-term ledger storage with BigTable

### PR DESCRIPTION
**Pull request by mvines**

### CODE REVIEW

File: core/Cargo.toml [hunk 1]
```diff
@@ -61,17 +61,18 @@ solana-runtime = { path = "../runtime", version = "1.1.22" }
...
 solana-vote-signer = { path = "../vote-signer", version = "1.1.22" }
 solana-sys-tuner = { path = "../sys-tuner", version = "1.1.22" }
 tempfile = "3.1.0"
 thiserror = "1.0"
-tokio = "0.1"
-tokio-codec = "0.1"
-tokio-fs = "0.1"
-tokio-io = "0.1"
+tokio = { version = "0.2.22", features = ["full"] }
+tokio_01 = { version = "0.1", package = "tokio" }
```
Comment (garious) [inline core/Cargo.toml:73]: that makes me queasy

Comment (mvines) [reply core/Cargo.toml:73]: No doubt.   Until we boot the jsonrpc-* crates we're stuck with tokio 0.1 in the tree though -- they're using hyper 0.12.  Best we can we do now is port everything that doesn't touch jsonrpc-* to tokio 0.2

**Comment by mvines** [comment]
I loaded the first 100k slots into BigTable, looks good so far!

Genesis:
```
$ solana-ledger-tool bigtable block 0
Slot: 0
Parent Slot: 0
Blockhash: 4sGjMW1sUnHzSxGspuhpqLDx6wiyjNtZAMdL4VZHirAn
Previous Blockhash: 4sGjMW1sUnHzSxGspuhpqLDx6wiyjNtZAMdL4VZHirAn
```

Slot 1, which contains a vote from each of the four bootstrap validators on genesis:
```
$ solana-ledger-tool bigtable block 0
Slot: 1
Parent Slot: 0
Blockhash: AcknnkY4ok5BZuk69WijDETKVRUPZoMtniZHMe4ZaK1e
Previous Blockhash: 4sGjMW1sUnHzSxGspuhpqLDx6wiyjNtZAMdL4VZHirAn
Transaction 0:
  Recent Blockhash: 4sGjMW1sUnHzSxGspuhpqLDx6wiyjNtZAMdL4VZHirAn
  Signature 0: 39V8tR2Q8Ar3WwMBfVTRPFr7AakLHy5wp7skJNBL7ET6ARoikqc1TaMiuXEtHiNPLQKoeiVr5XnKH8QtjdonN4yM
  MessageHeader { num_required_signatures: 1, num_readonly_signed_accounts: 0, num_readonly_unsigned_accounts: 3 }
  Account 0: GdnSyH3YtwcxFvQrVVJMm1JhTS4QVX7MFsX56uJLUfiZ
  Account 1: sCtiJieP8B3SwYnXemiLpRFRR8KJLMtsMVN25fAFWjW
  Account 2: SysvarS1otHashes111111111111111111111111111
  Account 3: SysvarC1ock11111111111111111111111111111111
  Account 4: Vote111111111111111111111111111111111111111
  Instruction 0
    Program: Vote111111111111111111111111111111111111111 (4)
    Account 0: sCtiJieP8B3SwYnXemiLpRFRR8KJLMtsMVN25fAFWjW (1)
    Account 1: SysvarS1otHashes111111111111111111111111111 (2)
    Account 2: SysvarC1ock11111111111111111111111111111111 (3)
    Account 3: GdnSyH3YtwcxFvQrVVJMm1JhTS4QVX7MFsX56uJLUfiZ (0)
    Vote(Vote { slots: [0], hash: CWZbL5pkFWdEhdM81CVSx14DzGeGUiBAqWGXQQ7RgTv9, timestamp: Some(1584369051) })
  Status: Unavailable
Transaction 1:
  Recent Blockhash: 4sGjMW1sUnHzSxGspuhpqLDx6wiyjNtZAMdL4VZHirAn
  Signature 0: 4hPVA21e1KLQsEQkpSHk1UfkBBUfotajpkqESuV4tgqEdtEyDufaczAdZzSLexLhjytczDdSUFgwCTancgUWFzym
  MessageHeader { num_required_signatures: 1, num_readonly_signed_accounts: 0, num_readonly_unsigned_accounts: 3 }
  Account 0: CakcnaRDHka2gXyfbEd2d3xsvkJkqsLw2akB3zsN1D2S
  Account 1: 9bRDrYShoQ77MZKYTMoAsoCkU7dAR24mxYCBjXLpfEJx
  Account 2: SysvarS1otHashes111111111111111111111111111
  Account 3: SysvarC1ock11111111111111111111111111111111
  Account 4: Vote111111111111111111111111111111111111111
  Instruction 0
    Program: Vote111111111111111111111111111111111111111 (4)
    Account 0: 9bRDrYShoQ77MZKYTMoAsoCkU7dAR24mxYCBjXLpfEJx (1)
    Account 1: SysvarS1otHashes111111111111111111111111111 (2)
    Account 2: SysvarC1ock11111111111111111111111111111111 (3)
    Account 3: CakcnaRDHka2gXyfbEd2d3xsvkJkqsLw2akB3zsN1D2S (0)
    Vote(Vote { slots: [0], hash: CWZbL5pkFWdEhdM81CVSx14DzGeGUiBAqWGXQQ7RgTv9, timestamp: Some(1584369051) })
  Status: Unavailable
Transaction 2:
  Recent Blockhash: 4sGjMW1sUnHzSxGspuhpqLDx6wiyjNtZAMdL4VZHirAn
  Signature 0: FAWA66fudpiwdRDDQ4DRxdJsRvawvauwg4vQkm98ZHFpXmW5N7xzRiTRpt8QiZ19s1aVbzKgXW6kEZanwHeDFNS
  MessageHeader { num_required_signatures: 1, num_readonly_signed_accounts: 0, num_readonly_unsigned_accounts: 3 }
  Account 0: DE1bawNcRJB9rVm3buyMVfr8mBEoyyu73NBovf2oXJsJ
  Account 1: 8XgHUtBRY6qePVYERxosyX3MUq8NQkjtmFDSzQ2WpHTJ
  Account 2: SysvarS1otHashes111111111111111111111111111
  Account 3: SysvarC1ock11111111111111111111111111111111
  Account 4: Vote111111111111111111111111111111111111111
  Instruction 0
    Program: Vote111111111111111111111111111111111111111 (4)
    Account 0: 8XgHUtBRY6qePVYERxosyX3MUq8NQkjtmFDSzQ2WpHTJ (1)
    Account 1: SysvarS1otHashes111111111111111111111111111 (2)
    Account 2: SysvarC1ock11111111111111111111111111111111 (3)
    Account 3: DE1bawNcRJB9rVm3buyMVfr8mBEoyyu73NBovf2oXJsJ (0)
    Vote(Vote { slots: [0], hash: CWZbL5pkFWdEhdM81CVSx14DzGeGUiBAqWGXQQ7RgTv9, timestamp: Some(1584369051) })
  Status: Unavailable
Transaction 3:
  Recent Blockhash: 4sGjMW1sUnHzSxGspuhpqLDx6wiyjNtZAMdL4VZHirAn
  Signature 0: 58A6RFEk5AoFqXTtKfqLhLrs7mhJYWGQTZegoCXPZuneM6Spi47SNYk2M6d9MVzHbC9CpBVk5vrq24yyNgeQNK2p
  MessageHeader { num_required_signatures: 1, num_readonly_signed_accounts: 0, num_readonly_unsigned_accounts: 3 }
  Account 0: 7Np41oeYqPefeNQEHSv1UDhYrehxin3NStELsSKCT4K2
  Account 1: 4785anyR2rYSas6cQGHtykgzwYEtChvFYhcEgdDw3gGL
  Account 2: SysvarS1otHashes111111111111111111111111111
  Account 3: SysvarC1ock11111111111111111111111111111111
  Account 4: Vote111111111111111111111111111111111111111
  Instruction 0
    Program: Vote111111111111111111111111111111111111111 (4)
    Account 0: 4785anyR2rYSas6cQGHtykgzwYEtChvFYhcEgdDw3gGL (1)
    Account 1: SysvarS1otHashes111111111111111111111111111 (2)
    Account 2: SysvarC1ock11111111111111111111111111111111 (3)
    Account 3: 7Np41oeYqPefeNQEHSv1UDhYrehxin3NStELsSKCT4K2 (0)
    Vote(Vote { slots: [0], hash: CWZbL5pkFWdEhdM81CVSx14DzGeGUiBAqWGXQQ7RgTv9, timestamp: Some(1584369051) })
  Status: Unavailable
```

**Comment by mvines** [comment]
v1.1 is retired

**Pull request closed without merging**